src/zenodo_client/api.py

```diff
@@ -154,48 +165,89 @@ def publish(self, deposition_id: str, sleep: bool = True) -> requests.Response:
         res.raise_for_status()
         return res
 
-    def update(self, deposition_id: str, paths: Paths) -> requests.Response:
-        """Create a new version of the given record with the given files."""
-        # Prepare a new version based on the old version
-        # see: https://developers.zenodo.org/#new-version)
-        res = requests.post(
-            f"{self.depositions_base}/{deposition_id}/actions/newversion",
-            params={"access_token": self.access_token},
-        )
-        res.raise_for_status()
+    def update(
```

Comment:
this function's job is to update the files in a record. I would rather you take all of the new changes that are related to updating the data in a record, and make a new function.

Reply:
Sure, can do: will remove the parameter `data` and think about a new function. How does `update_metadata()` sound?

Would you also rather keep the function parameters as they are, or are you fine with a new (backwards compatible) option `publish` ?

If files are updated in an existing record, then there will always be a new version, so I can remove `new_version`.

Reply:
update_metadata is great. I'm fine to add the boolean option to turn off publishing for both of these functions